File: plugin/testData/shared/basics/Typealias.shared.kt
package com.test

typealias MyInteger = Int

class Thing()

typealias MyThing = Thing
typealias MyList<T> = List<T>
typealias ListOfThings = MyList<MyThing>

fun main(){
    val stuff: ListOfThings = listOf(MyThing(), Thing())
    println("Success")
}

File: plugin/testDataOutputTs/shared/basics/Typealias.shared.ts
//Package: com.test
//Converted using Khrysalis2


export type MyInteger = number
export let MyInteger = number
 
 

class Thing {
    
    public constructor() {
    }
    
}
 
 

export type MyThing = Thing
export let MyThing = Thing
 

export type MyList< T > = Array<T>
export let MyList = Array
 

export type ListOfThings = MyList<MyThing>
export let ListOfThings = MyList
 
 

export function main(): void {
    const stuff : ListOfThings = listOf(new MyThing(), new Thing());
    console.log("Success")
}
 
